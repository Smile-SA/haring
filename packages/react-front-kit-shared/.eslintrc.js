module.exports = {
  extends: ['custom'],
  parserOptions: {
    project: ['./tsconfig.json'],
    tsconfigRootDir: __dirname,
  },
  root: true,
};
