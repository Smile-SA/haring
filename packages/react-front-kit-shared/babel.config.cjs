const { babelConfig } = require('test');

module.exports = babelConfig;
