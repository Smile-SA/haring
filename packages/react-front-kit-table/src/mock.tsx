/* eslint-disable react-refresh/only-export-components */
export * from './Components/Table/Table.mock';
export * from './Components/TableGridView/TableGridView.mock';
