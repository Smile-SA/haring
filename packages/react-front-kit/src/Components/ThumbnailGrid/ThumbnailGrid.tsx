'use client';

import type { IThumbnail } from '../Thumbnail/Thumbnail';
import type { SimpleGridProps } from '@mantine/core';
import type {
  IAction,
  IActionConfirmModalProps,
} from '@smile/react-front-kit-shared';
import type { ReactElement } from 'react';

import { Button, Group, SimpleGrid } from '@mantine/core';
import { createStyles } from '@mantine/styles';
import { useState } from 'react';

import { ConfirmModal } from '../ConfirmModal/ConfirmModal';
import { Thumbnail } from '../Thumbnail/Thumbnail';

const useStyles = createStyles((theme) => ({
  container: {
    display: 'flex',
    flexDirection: 'column',
    gap: 24,
  },
  topBar: {
    alignItems: 'center',
    background: theme.fn.primaryColor(),
    borderRadius: 4,
    color: 'white',
    display: 'inline-flex',
    justifyContent: 'space-between',
    padding: '16px 24px',
  },
}));

function defaultSelectedElementsText(n: number): string {
  return `${n} selected file${n > 1 ? 's' : ''}`;
}

type IGridAction = IAction<IThumbnail[]>;
type IGridActionConfirmModalProps = IActionConfirmModalProps<IThumbnail[]>;

export interface IThumbnailGridProps extends SimpleGridProps {
  gridActions?: IGridAction[];
  onThumbnailClick?: (item: IThumbnail, index: number) => void;
  selectedElementsText?: (numberOfSelectedElements: number) => string;
  thumbnails: IThumbnail[];
}

/** Additional props will be forwarded to the [Mantine SimpleGrid component](https://mantine.dev/core/simple-grid) */
export function ThumbnailGrid(props: IThumbnailGridProps): ReactElement {
  const {
    gridActions = [],
    onThumbnailClick,
    selectedElementsText = defaultSelectedElementsText,
    thumbnails,
    ...simpleGridProps
  } = props;

  const selectedElements = thumbnails.filter((thumbnail) => thumbnail.selected);
  const numberOfSelectedElements = selectedElements.length;
  const [confirmAction, setConfirmAction] =
    useState<IGridActionConfirmModalProps | null>(null);

  const { classes } = useStyles();

  function handleSelect(index: number): void {
    onThumbnailClick?.(thumbnails[index], index);
  }

  function setModal(action: IGridAction): void {
    setConfirmAction({
      cancelColor: action.confirmModalProps?.cancelColor,
      cancelLabel: action.confirmModalProps?.cancelLabel,
      children: action.confirmModalProps?.children,
      confirmColor: action.confirmModalProps?.confirmColor,
      confirmLabel: action.confirmModalProps?.confirmLabel,
      onConfirm: () => action.onAction?.(selectedElements),
      title: action.confirmModalProps?.title,
    });
  }

  function handleGridAction(action: IGridAction): void {
    if (action.confirmation) {
      setModal(action);
    } else {
      action.onAction?.(selectedElements);
    }
  }

  function clearConfirmAction(): void {
    setConfirmAction(null);
  }

  function handleClose(): void {
    clearConfirmAction();
  }

  function handleModalButton(onAction?: (item: IThumbnail[]) => void): void {
    onAction?.(selectedElements);
    handleClose();
  }

  return (
    <>
      <div className={classes.container}>
        {numberOfSelectedElements > 0 && (
          <div className={classes.topBar}>
            <span>{selectedElementsText(numberOfSelectedElements)}</span>
            {gridActions.length > 0 && (
              <Group>
                {gridActions.map((action) => (
                  <Button
                    key={action.id}
                    color={action.color}
                    leftIcon={action.icon}
                    onClick={() => handleGridAction(action)}
                    variant={action.color ? 'filled' : 'default'}
                  >
                    {action.label}
                  </Button>
                ))}
              </Group>
            )}
          </div>
        )}
        <SimpleGrid {...simpleGridProps}>
          {thumbnails.map((thumbnail, index) => (
            <Thumbnail
              key={thumbnail.id}
              onClick={() => handleSelect(index)}
              {...thumbnail}
            />
          ))}
        </SimpleGrid>
      </div>
      <ConfirmModal
        {...confirmAction}
        onCancel={() => handleModalButton(confirmAction?.onCancel)}
        onClose={handleClose}
        onConfirm={() => handleModalButton(confirmAction?.onConfirm)}
        opened={Boolean(confirmAction)}
      >
        {confirmAction?.children}
      </ConfirmModal>
    </>
  );
}
