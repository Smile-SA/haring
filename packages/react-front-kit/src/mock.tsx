/* eslint-disable react-refresh/only-export-components */

export * from './Components/ResponsiveTabs/ResponsiveTabs.mock';
export * from './Components/SidebarMenu/SidebarMenu.mock';
export * from './Components/SwitchableView/SwitchableView.mock';
export * from './Components/Thumbnail/Thumbnail.mock';
export * from './Components/ThumbnailGrid/ThumbnailGrid.mock';
export * from './Layouts/FoldableColumnLayout/FoldableColumnLayout.mock';
