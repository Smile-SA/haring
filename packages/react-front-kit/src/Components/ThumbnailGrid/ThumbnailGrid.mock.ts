import type { IThumbnail } from '../Thumbnail/Thumbnail';

import { thumbnailActions } from '../Thumbnail/Thumbnail.mock';

export const baseThumbnail: Omit<IThumbnail, 'id' | 'label'> = {
  actions: thumbnailActions,
  iconType: 'PDF',
};

export const thumbnails: IThumbnail[] = [
  {
    id: '1',
    label: 'Debit_Suivi_PREV',
    ...baseThumbnail,
  },
  {
    id: '2',
    label: 'Debit_Suivi_PREV_2',
    ...baseThumbnail,
    selected: true,
  },
  {
    id: '3',
    label: 'Debit_Suivi_PREV_3',
    ...baseThumbnail,
  },
];

export function getModalTitle(n: number): string {
  return `Remove ${n} file${n > 1 ? 's' : ''}`;
}
