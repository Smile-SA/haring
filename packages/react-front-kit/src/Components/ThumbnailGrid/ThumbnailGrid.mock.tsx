import type { IThumbnail } from '../Thumbnail/Thumbnail';
import type { HandlerFunction } from '@storybook/addon-actions';

import { Trash } from '@phosphor-icons/react';
import { FolderMove } from '@smile/react-front-kit-shared';
import { action } from '@storybook/addon-actions';

import { thumbnailActionsMock } from '../Thumbnail/Thumbnail.mock';

export const baseThumbnailMock: Omit<IThumbnail, 'id' | 'label'> = {
  actions: thumbnailActionsMock,
  iconType: 'PDF',
};

export const thumbnails: IThumbnail[] = [
  {
    id: '1',
    label: 'Debit_Suivi_PREV',
    ...baseThumbnailMock,
  },
  {
    id: '2',
    label: 'Debit_Suivi_PREV_2',
    ...baseThumbnailMock,
    selected: true,
  },
  {
    id: '3',
    label: 'Debit_Suivi_PREV_3',
    ...baseThumbnailMock,
  },
];

export const thumbnailGridMock = {
  cols: 5,
  gridActions: [
    {
      icon: <FolderMove size={16} />,
      id: 'move',
      label: 'Move in tree',
      onAction: action('Move selected in tree'),
    },
    {
      color: 'red',
      confirmModalProps: {
        cancelLabel: 'Abort',
        children: <p>Are you sure ?</p>,
        confirmColor: 'red',
        confirmLabel: 'Remove',
        title: 'remove x files ?',
      },
      confirmation: true,
      icon: <Trash size={16} />,
      id: 'delete',
      label: 'Delete',
      onAction: action('Delete selected'),
    },
  ],
  onThumbnailClick: (): HandlerFunction => action('Thumbnail clicked'),
  spacing: 25,
  thumbnails,
  verticalSpacing: 25,
};
