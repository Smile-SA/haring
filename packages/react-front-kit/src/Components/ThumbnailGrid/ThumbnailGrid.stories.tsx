import type { Meta, StoryObj } from '@storybook/react';

import { Trash } from '@phosphor-icons/react';
import { FolderMove } from '@smile/react-front-kit-shared';
import { action } from '@storybook/addon-actions';

import { ThumbnailGrid as Cmp } from './ThumbnailGrid';
import { getModalTitle, thumbnails } from './ThumbnailGrid.mock';

const meta = {
  component: Cmp,
  tags: ['autodocs'],
  title: '3-custom/Components/ThumbnailGrid',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const ThumbnailGrid: IStory = {
  args: {
    cols: 5,
    gridActions: [
      {
        icon: <FolderMove size={16} />,
        id: 'move',
        label: 'Move in tree',
        onAction: action('Move selected in tree'),
      },
      {
        color: 'red',
        confirmModalProps: {
          cancelLabel: 'Abort',
          children: <p>Are you sure ?</p>,
          confirmColor: 'red',
          confirmLabel: 'Remove',
          title: getModalTitle(
            thumbnails.filter((thumbnail) => thumbnail.selected).length,
          ),
        },
        confirmation: true,
        icon: <Trash size={16} />,
        id: 'delete',
        label: 'Delete',
        onAction: action('Delete selected'),
      },
    ],
    onThumbnailClick: () => action('Thumbnail clicked'),
    spacing: 25,
    thumbnails,
    verticalSpacing: 25,
  },
};
