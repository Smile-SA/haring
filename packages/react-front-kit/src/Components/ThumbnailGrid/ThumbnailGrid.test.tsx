import { renderWithProviders } from '@smile/react-front-kit-shared/test-utils';

import { ThumbnailGrid } from './ThumbnailGrid';
import { thumbnailGridActionsMock, thumbnails } from './ThumbnailGrid.mock';

describe('ThumbnailGrid', () => {
  beforeEach(() => {
    // Prevent mantine random ID
    Math.random = () => 0.42;
  });
  it('matches snapshot', () => {
    const { container } = renderWithProviders(
      <ThumbnailGrid
        actions={thumbnailGridActionsMock}
        thumbnails={thumbnails}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
