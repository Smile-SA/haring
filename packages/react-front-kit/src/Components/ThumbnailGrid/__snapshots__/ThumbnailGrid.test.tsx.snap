// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThumbnailGrid matches snapshot 1`] = `
<div>
  <div
    class="mantine-1skbp38"
  >
    <div
      class="___ref-actionBar mantine-15edn1r"
    >
      <span>
        1 selected file
      </span>
      <div
        class="mantine-Group-root mantine-1vlub25"
      >
        <button
          class="mantine-UnstyledButton-root mantine-Button-root mantine-1fg4bc4"
          data-button="true"
          type="button"
        >
          <div
            class="mantine-1wpc1xj mantine-Button-inner"
          >
            <span
              class="mantine-Button-icon mantine-Button-leftIcon mantine-1696ey6"
            >
              <svg
                fill="none"
                height="16"
                stroke="currentColor"
                viewBox="0 0 12 12"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3.58333 10H1.41667C1.30616 10 1.20018 9.95564 1.12204 9.87668C1.0439 9.79771 1 9.69062 1 9.57895V2.42105C1 2.30938 1.0439 2.20229 1.12204 2.12332C1.20018 2.04436 1.30616 2 1.41667 2H4.19271C4.2828 2.00037 4.37041 2.02988 4.44271 2.08421L5.89062 3.17895C5.96292 3.23327 6.05053 3.26278 6.14062 3.26316H10.5833C10.6938 3.26316 10.7998 3.30752 10.878 3.38648C10.9561 3.46544 11 3.57254 11 3.68421V5.36842"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
                <path
                  d="M6 9H11"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
                <path
                  d="M9 7L11 9L9 11"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </span>
            <span
              class="mantine-Button-label mantine-1e30p2e"
            >
              Move in tree
            </span>
          </div>
        </button>
        <button
          class="mantine-UnstyledButton-root mantine-Button-root mantine-3a0nxt"
          data-button="true"
          type="button"
        >
          <div
            class="mantine-1wpc1xj mantine-Button-inner"
          >
            <span
              class="mantine-Button-icon mantine-Button-leftIcon mantine-1696ey6"
            >
              <svg
                fill="currentColor"
                height="16"
                viewBox="0 0 256 256"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M216,48H176V40a24,24,0,0,0-24-24H104A24,24,0,0,0,80,40v8H40a8,8,0,0,0,0,16h8V208a16,16,0,0,0,16,16H192a16,16,0,0,0,16-16V64h8a8,8,0,0,0,0-16ZM96,40a8,8,0,0,1,8-8h48a8,8,0,0,1,8,8v8H96Zm96,168H64V64H192ZM112,104v64a8,8,0,0,1-16,0V104a8,8,0,0,1,16,0Zm48,0v64a8,8,0,0,1-16,0V104a8,8,0,0,1,16,0Z"
                />
              </svg>
            </span>
            <span
              class="mantine-Button-label mantine-1e30p2e"
            >
              Delete
            </span>
          </div>
        </button>
      </div>
    </div>
    <div
      class="mantine-SimpleGrid-root mantine-17ynw9a"
    >
      <div
        class="mantine-af86sk"
      >
        <div
          class="mantine-Group-root mantine-p1wipv"
        >
          <div
            class="mantine-1r83wdq"
          >
            <svg
              class="mantine-1qu33rh"
              fill="#0b7285"
              height="22"
              viewBox="0 0 256 256"
              width="22"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M200,164v8h12a12,12,0,0,1,0,24H200v12a12,12,0,0,1-24,0V152a12,12,0,0,1,12-12h32a12,12,0,0,1,0,24ZM92,172a32,32,0,0,1-32,32H56v4a12,12,0,0,1-24,0V152a12,12,0,0,1,12-12H60A32,32,0,0,1,92,172Zm-24,0a8,8,0,0,0-8-8H56v16h4A8,8,0,0,0,68,172Zm100,8a40,40,0,0,1-40,40H112a12,12,0,0,1-12-12V152a12,12,0,0,1,12-12h16A40,40,0,0,1,168,180Zm-24,0a16,16,0,0,0-16-16h-4v32h4A16,16,0,0,0,144,180ZM36,108V40A20,20,0,0,1,56,20h96a12,12,0,0,1,8.49,3.52l56,56A12,12,0,0,1,220,88v20a12,12,0,0,1-24,0v-4H148a12,12,0,0,1-12-12V44H60v64a12,12,0,0,1-24,0ZM160,57V80h23Z"
              />
            </svg>
            <h3
              class="mantine-Text-root mantine-1oer4ck"
            >
              Debit_Suivi_PREV
            </h3>
          </div>
          <div>
            <button
              aria-controls="mantine-f4bipx4bi-dropdown"
              aria-expanded="false"
              aria-haspopup="menu"
              class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-j5kg09"
              id="mantine-f4bipx4bi-target"
              type="button"
            >
              <div>
                <svg
                  class="mantine-pusk2m"
                  fill="currentColor"
                  height="16"
                  viewBox="0 0 256 256"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M112,60a16,16,0,1,1,16,16A16,16,0,0,1,112,60Zm16,52a16,16,0,1,0,16,16A16,16,0,0,0,128,112Zm0,68a16,16,0,1,0,16,16A16,16,0,0,0,128,180Z"
                  />
                </svg>
              </div>
            </button>
          </div>
        </div>
        <div
          class="mantine-Image-root mantine-yxmaw9"
          style="width: 100%;"
        >
          <figure
            class="mantine-qenwvq mantine-Image-figure"
          >
            <div
              class="mantine-1iugybl mantine-Image-imageWrapper"
            >
              <img
                class="mantine-1bi35m7 mantine-Image-image"
                src=""
                style="object-fit: cover; width: 100%; height: auto;"
              />
            </div>
          </figure>
        </div>
      </div>
      <div
        class="mantine-1ygkvb0"
      >
        <div
          class="mantine-Group-root mantine-p1wipv"
        >
          <div
            class="mantine-1r83wdq"
          >
            <svg
              class="mantine-1qu33rh"
              fill="#f1f3f5"
              height="22"
              viewBox="0 0 256 256"
              width="22"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M200,164v8h12a12,12,0,0,1,0,24H200v12a12,12,0,0,1-24,0V152a12,12,0,0,1,12-12h32a12,12,0,0,1,0,24ZM92,172a32,32,0,0,1-32,32H56v4a12,12,0,0,1-24,0V152a12,12,0,0,1,12-12H60A32,32,0,0,1,92,172Zm-24,0a8,8,0,0,0-8-8H56v16h4A8,8,0,0,0,68,172Zm100,8a40,40,0,0,1-40,40H112a12,12,0,0,1-12-12V152a12,12,0,0,1,12-12h16A40,40,0,0,1,168,180Zm-24,0a16,16,0,0,0-16-16h-4v32h4A16,16,0,0,0,144,180ZM36,108V40A20,20,0,0,1,56,20h96a12,12,0,0,1,8.49,3.52l56,56A12,12,0,0,1,220,88v20a12,12,0,0,1-24,0v-4H148a12,12,0,0,1-12-12V44H60v64a12,12,0,0,1-24,0ZM160,57V80h23Z"
              />
            </svg>
            <h3
              class="mantine-Text-root mantine-1oer4ck"
            >
              Debit_Suivi_PREV_2
            </h3>
          </div>
          <div>
            <button
              aria-controls="mantine-f4bipx4bi-dropdown"
              aria-expanded="false"
              aria-haspopup="menu"
              class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-1w8jhtw"
              id="mantine-f4bipx4bi-target"
              type="button"
            >
              <div>
                <svg
                  class="mantine-pusk2m"
                  fill="currentColor"
                  height="16"
                  viewBox="0 0 256 256"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M112,60a16,16,0,1,1,16,16A16,16,0,0,1,112,60Zm16,52a16,16,0,1,0,16,16A16,16,0,0,0,128,112Zm0,68a16,16,0,1,0,16,16A16,16,0,0,0,128,180Z"
                  />
                </svg>
              </div>
            </button>
          </div>
        </div>
        <div
          class="mantine-Image-root mantine-yxmaw9"
          style="width: 100%;"
        >
          <figure
            class="mantine-qenwvq mantine-Image-figure"
          >
            <div
              class="mantine-1iugybl mantine-Image-imageWrapper"
            >
              <img
                class="mantine-1bi35m7 mantine-Image-image"
                src=""
                style="object-fit: cover; width: 100%; height: auto;"
              />
            </div>
          </figure>
        </div>
      </div>
      <div
        class="mantine-af86sk"
      >
        <div
          class="mantine-Group-root mantine-p1wipv"
        >
          <div
            class="mantine-1r83wdq"
          >
            <svg
              class="mantine-1qu33rh"
              fill="#0b7285"
              height="22"
              viewBox="0 0 256 256"
              width="22"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M200,164v8h12a12,12,0,0,1,0,24H200v12a12,12,0,0,1-24,0V152a12,12,0,0,1,12-12h32a12,12,0,0,1,0,24ZM92,172a32,32,0,0,1-32,32H56v4a12,12,0,0,1-24,0V152a12,12,0,0,1,12-12H60A32,32,0,0,1,92,172Zm-24,0a8,8,0,0,0-8-8H56v16h4A8,8,0,0,0,68,172Zm100,8a40,40,0,0,1-40,40H112a12,12,0,0,1-12-12V152a12,12,0,0,1,12-12h16A40,40,0,0,1,168,180Zm-24,0a16,16,0,0,0-16-16h-4v32h4A16,16,0,0,0,144,180ZM36,108V40A20,20,0,0,1,56,20h96a12,12,0,0,1,8.49,3.52l56,56A12,12,0,0,1,220,88v20a12,12,0,0,1-24,0v-4H148a12,12,0,0,1-12-12V44H60v64a12,12,0,0,1-24,0ZM160,57V80h23Z"
              />
            </svg>
            <h3
              class="mantine-Text-root mantine-1oer4ck"
            >
              Debit_Suivi_PREV_3
            </h3>
          </div>
          <div>
            <button
              aria-controls="mantine-f4bipx4bi-dropdown"
              aria-expanded="false"
              aria-haspopup="menu"
              class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-j5kg09"
              id="mantine-f4bipx4bi-target"
              type="button"
            >
              <div>
                <svg
                  class="mantine-pusk2m"
                  fill="currentColor"
                  height="16"
                  viewBox="0 0 256 256"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M112,60a16,16,0,1,1,16,16A16,16,0,0,1,112,60Zm16,52a16,16,0,1,0,16,16A16,16,0,0,0,128,112Zm0,68a16,16,0,1,0,16,16A16,16,0,0,0,128,180Z"
                  />
                </svg>
              </div>
            </button>
          </div>
        </div>
        <div
          class="mantine-Image-root mantine-yxmaw9"
          style="width: 100%;"
        >
          <figure
            class="mantine-qenwvq mantine-Image-figure"
          >
            <div
              class="mantine-1iugybl mantine-Image-imageWrapper"
            >
              <img
                class="mantine-1bi35m7 mantine-Image-image"
                src=""
                style="object-fit: cover; width: 100%; height: auto;"
              />
            </div>
          </figure>
        </div>
      </div>
    </div>
  </div>
</div>
`;
